apps/network/tests/database/test_user.py

```diff
@@ -0,0 +1,13 @@
+import pytest
+
+from src.users.user import User
+from .presets.user import user_metrics
+
+
+@pytest.mark.parametrize(
+    ("email, hashed_password, salt, role"), user_metrics,
+)
+def test_create_role_object(email, hashed_password, salt, role, database):
+    user = User(email=email, hashed_password=hashed_password, salt=salt, role=role)
```

Comment:
Ideally, the user will never be submitting the hashed password and salt, nor would they submit a role. Rather, the role will be defaultly assigned to their account (unless created by a high-ranking user). We should consider having the ability to test submitting an email and plaintext password, and then have it save with the correctly hashed password, salt, and role applied.

Reply:
Do you think this falls outside the scope of this PR ? At first I see such an improvement well aligned with the PR that adds the related routes,  because we also put together the logic that does it in that PR.

Reply:
Sure, I suppose for this issue/PR that's fair. But you'll be doing this API work in your next 3-4 issues. You can do it then, just set a reminder to go back and change this.